client/sites-dashboard-v2/hooks/use-sync-selected-site-feature.ts

```diff
@@ -0,0 +1,63 @@
+import page from '@automattic/calypso-router';
+import { useEffect, useState } from 'react';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { DataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces';
+
+export function useSyncSelectedSiteFeature( {
```

Comment:
I've basically skipped this part because it has some kind of complexity, and it's Dotcom URL/route sync/build.